src/app/branded/step-2/branded-checkout-step-2.spec.js

```diff
@@ -21,20 +21,28 @@ describe('branded checkout step 2', () => {
     it('should load cart', () => {
       jest.spyOn($ctrl, 'loadCart').mockImplementation(() => {})
       jest.spyOn($ctrl, 'loadRadioStation').mockImplementation(() => {})
+      jest.spyOn($ctrl.brandedAnalyticsFactory, 'reviewOrder')
       $ctrl.$onInit()
 
       expect($ctrl.loadCart).toHaveBeenCalled()
       expect($ctrl.loadRadioStation).toHaveBeenCalled()
+      expect($ctrl.brandedAnalyticsFactory.reviewOrder).toHaveBeenCalled()
     })
   })
 
   describe('loadCart', () => {
+    beforeEach(() => {
+      jest.spyOn($ctrl.brandedAnalyticsFactory, 'addPaymentInfo')
+    })
+
     it('should load cart data', () => {
-      jest.spyOn($ctrl.cartService, 'get').mockReturnValue(Observable.of('some data'))
+      const cartData = { items: [] }
+      jest.spyOn($ctrl.cartService, 'get').mockReturnValue(Observable.of(cartData))
       $ctrl.loadCart()
 
-      expect($ctrl.cartData).toEqual('some data')
+      expect($ctrl.cartData).toEqual(cartData)
       expect($ctrl.errorLoadingCart).toEqual(false)
+      expect($ctrl.brandedAnalyticsFactory.addPaymentInfo).toHaveBeenCalled(cartData.items[0])
```

Comment:
What about doing a test with the cover fee?

Reply:
Good catch. `toHaveBeenCalled` should have been `toHaveBeenCalledWith` also.